usmqe_tests/conftest.py

```diff
@@ -35,25 +38,41 @@ def get_name(fname):
     return fname[5:].replace('_', ' ')
 
 
-def measure_operation(operation):
+def measure_operation(
+        operation, minimal_time=None, metadata=None, measure_after=False):
     """
     Get dictionary with keys 'start', 'end' and 'result' that contain
     information about start and stop time of given function and its result.
 
     Args:
-        operation (function): Function to be performed.
+        operation (function): function to be performed.
+        minimal_time (int): minimal number of seconds to run, it can be more
+            based on given operation
+        metadata (dict): this can contain dictionary object with information
+            relevant to test (e.g. volume name, operating host, ...)
+        measure_after (bool): determine if time measurement is done before or
+            after the operation returns its state
```

Comment:
May I ask what is the purpose of measuring the time from the point when the operation returns?

Reply:
This is done because you can now specify moment to measure data after the setup is ready and returned data from graphite are constant and you don't have to handle that all data are not set up and not ready when the timer for the setup runs out. For example without this if you want to measure disk utilization without setting `measure_after`:

- you start to measure utilization,
- then you run `dd` command to fill disks,
- then you leave it some time to measure (or not if disk is not utilized before timer runs out)
- then you stop measurement and evaluate results.
- In results you have to handle that in each time frame should be different value of utilization.

To measure the process of loading data into disk would be a good testcase but I would add it in future.

Reply:
Ah, ok.